nixos/modules/services/networking/rosenpass.nix

```diff
@@ -0,0 +1,220 @@
+{ config
+, lib
+, options
+, pkgs
+, ...
+}:
+with builtins;
+with lib; let
+  cfg = config.services.rosenpass;
+  opt = options.services.rosenpass;
+  settingsFormat = pkgs.formats.toml { };
+in
+{
+  options.services.rosenpass = with types; {
+    enable = mkEnableOption "Rosenpass";
+
+    package = mkPackageOption pkgs "rosenpass" { };
+
+    defaultDevice = mkOption {
+      type = nullOr str;
+      description = "Name of the network interface to use for all peers by default.";
+      example = "wg0";
```

Comment:
Shouldn't this have a default if it can be null?

Reply:
No. If you don't set it, and you don't set `device` on some peer, then that should error, and not use a device of `null`.